tests/session/test_env_select.py

```diff
@@ -15,6 +15,57 @@
 
 CURRENT_PY_ENV = f"py{sys.version_info[0]}{sys.version_info[1]}"  # e.g. py310
 
+####################################################################
+#   Tests directly on env_select functionality.
+
+
+@pytest.mark.parametrize(
+    ("userinput", "envnames", "isall", "isdefault"),
+    [
+        (None, (), False, True),
+        ("", (), False, True),
+        ("a1", ("a1",), False, False),
+        ("a1,b2,c3", ("a1", "b2", "c3"), False, False),
+        #   If the user gives "ALL" as any envname, this becomes an "is_all" and other envnames are ignored.
+        ("ALL", (), True, False),
+        ("a1,ALL,b2", (), True, False),
+        #   Zero-length envnames are ignored as being not present. This is not intentional.
+        (",,a1,,,b2,,", ("a1", "b2"), False, False),
+        (",,", (), False, True),
+        #   Environment names with "invalid" characters are accepted here; the client is expected to deal with this.
+        ("\x01.-@\x02,xxx", ("\x01.-@\x02", "xxx"), False, False),
+    ],
+)
+def test_clienv(userinput: str, envnames: tuple[str], isall: bool, isdefault: bool) -> None:
```

Comment:
Please use the snake case for variable names.

Reply:
Done.